src/pages/studyView/filters/groupLogic/GroupLogic.tsx

```diff
@@ -21,7 +21,7 @@ export class GroupLogic extends React.Component<IGroupLogicProps, {}> {
         }
         _.reduce(this.props.components, (acc, next, index): any => {
                 if (index > 0) {
-                    acc.push(<span key={this.props.operation} className={styles.text}>{this.props.operation}</span>);
+                    acc.push(<span key={index} className={styles.text}>{this.props.operation}</span>);
```

Comment:
how come you switched from operation?  you should not use index as key.  it's probably ok here, but could be dangerous elsewhere